src/storage/v2/inmemory/storage.cpp

```diff
@@ -2196,14 +2204,13 @@ void InMemoryStorage::CollectGarbage(std::unique_lock<utils::ResourceLock> main_
       guard.unlock();
       // if lucky, there are no active transactions, hence nothing looking at the deltas
       // remove them now
-
-      auto const sum_func = [](uint32_t curr, auto const &gc_deltas) { return curr + gc_deltas.deltas_.size(); };
-      uint32_t const total_delta_size =
-          std::accumulate(unlinked_undo_buffers.begin(), unlinked_undo_buffers.end(), 0U, sum_func);
+      auto const released_delta_count =
+          std::accumulate(unlinked_undo_buffers.cbegin(), unlinked_undo_buffers.cend(), uint64_t{0},
+                          [](uint64_t curr, GCDeltas const &gc_deltas) { return curr + gc_deltas.deltas_.size(); });
 
       // Now total_deltas contains the sum of all deltas in the unlinked_undo_buffers list
       unlinked_undo_buffers.clear();
-      memgraph::metrics::DecrementCounter(memgraph::metrics::UnreleasedDeltaObjects, total_delta_size);
+      memgraph::metrics::DecrementCounter(memgraph::metrics::UnreleasedDeltaObjects, released_delta_count);
```

Comment:
Leaving this comment at a random line.
We are missing any metrics testing. Do you want to add it as part of this PR?